torch/_dynamo/variables/builtin.py

```diff
@@ -1073,31 +1073,33 @@ def call_map(self, tx, fn, seq):
             items = [fn.call_function(tx, [x], {}) for x in seq.unpack_var_sequence(tx)]
             return variables.TupleVariable(items)
 
-    def call_sum(self, tx, seq, **kwargs):
+    def call_sum(self, tx, seq, start=_SENTINEL):
```

Comment:
should be `self._SENTINEL`

Reply:
Please add a test that exercises the default value.

Reply:
@kit1980 This should have been exercised by our test suite.

Reply:
Can it reference `self` in other arguments, at best it should be `BuiltinVariable._SENTINEL`, right?

Reply:
@kit1980 Could you provide a unit test that fails here? It's a class attribute so it should be defined in that context too, unless I am missing something?

Reply:
This PR could have had a better description, it feels wrong to reduce all *kwargs to just `start`? (though according to spec this is probably fine)

Reply:
Sorry, not `self._SENTINEL`, but `BuiltinVariable._SENTINEL`

Reply:
@Skylion007 several thousands tests failed internally.
For example, test/inductor:binary_folding.

Reply:
Maybe the way we run test internally affect the scoping? Using BuiltinVariable._SENTINEL should be an easy fix/workaround to try.

Reply:
True, but it's not failing on external CI so tough to verify on our end.

Reply:
Sorry for the noise, PRs were imported out of order, this one should have been imported after https://github.com/pytorch/pytorch/pull/116398 that introduced SENTINEL in the first place

Reply:
The methods are created in the class namespace.

```python
class Foo:
    SENTINEL = object()

    def my_method(self, x=SENTINEL):
        assert x is self.SENTINEL

Foo().my_method()
```

While defining `my_method`, it can reference `SENTINEL` as the default value in the same scope (a temporary namespace during class creation). Also, we cannot access `Foo.SENTINEL` during the class creation because class `Foo` has not been created yet (will raise `NameError` for the name `Foo` is not defined).

In the method body, we should use `Foo.SENTINEL` or `self.SENTINEL`. Because the method body runs under its separate scope and the class is already created.

The class creation is roughly equivalent to: 

```python
namespace Foo {

SENTINEL = object()

def my_method(self, x=SENTINEL):  # no Foo.SENTINEL or self.SENTINEL here in function signature
    assert x is self.SENTINEL     # should use Foo.SENTINEL or self.SENTINEL in the function body

} # end namespace Foo

# assign attributes in namespace to class Foo
Foo = type('Foo', (object,), {})  # create an empty class
Foo.SENTINEL = Foo::SENTINEL
Foo.my_method = MethodType(Foo::my_method)
```

We can move the sentinel variable to the global scope to reduce misunderstandings:

```diff
+ _SENTINEL = object()

  class BuiltinVariable:
-     _SENTINEL = object()

      def call_sum(self, tx, seq, start=_SENTINEL):
          ...
-         if start is self._SENTINEL:
+         if start is _SENTINEL:
              ...
```

Reply:
[XuehaiPan](https://github.com/XuehaiPan) Everything's fine as is, don't worry. It was my misunderstanding triggered by some tools issues on the Meta side.